packages/headless/src/features/case-assist/case-assist-analytics-selectors.ts

```diff
@@ -0,0 +1,124 @@
+import {CaseAssistAppState} from '../../state/case-assist-app-state';
+
+export enum KnownCaseFields {
+  id = 'id',
+  subject = 'subject',
+  description = 'description',
+  category = 'category',
+  productId = 'productId',
+}
+
+const isCustomFieldName = (fieldName: string) =>
+  !(Object.values(KnownCaseFields) as string[]).includes(fieldName);
+
+export const selectCase = (state: Partial<CaseAssistAppState>) => {
+  const selectFieldValue = (fieldName: string) =>
+    selectCaseInputValue(state, fieldName) ??
+    selectCaseFieldValue(state, fieldName);
+
+  return {
+    id: selectFieldValue(KnownCaseFields.id),
+    subject: selectFieldValue(KnownCaseFields.subject),
+    description: selectFieldValue(KnownCaseFields.description),
+    category: selectFieldValue(KnownCaseFields.category),
+    productId: selectFieldValue(KnownCaseFields.productId),
+    custom: {
+      ...selectCustomCaseInputValues(state),
+      ...selectCustomCaseFieldValues(state),
+    },
+  };
+};
+
+export const selectCaseInputValue = (
```

Comment:
I'm unsure about the naming scheme here. These are selectors but they sound like actions. I guess "select" is technically correct for a "selector" :P but in the case of `selectCaseClassification` I might be confused.

Reply:
I notice that the pattern in some of our other files (such as `pagination-selectors.ts`) is `*Selector()`, I would prefer that to avoid confusion.

Reply:
I see the numeric facet also has some selectors and it is using the `Selector` suffix instead. I'll use the same convention.

Reply:
Thanks!